### PR TITLE
Add Process Managers

### PR DESCRIPTION
**Pull request by Levi-Armstrong**
- [x] Fix broken unit tests
- [ ] Implement the task validators for 
  - [ ] Format check (for checking that seed matches instruction)
  - [ ] Discrete Collision
  - [ ] Continous collision
  - [ ] etc
- [x] Implement the rest of the step generators
  - [x] Implement fixed size joint interpolate generators
  - [x] Implement fixed size assign generators
  - [ ] Implement longest valid segment generators
- [ ] Address required todos placed everywhere in the code.
  - [x] tesseract_motion_planners
  - [ ] tesseract_process_managers
- [ ] Unit tests. I tried to structure everything such that components are easily testable. However, not many of them are currently tested. 
  - [x] Add unit test for fixed size joint interpolate generators
  - [ ] Add unit test for fixed size assign generators
  - [ ] Add unit test for longest valid segment generators
  - [ ] Add unit test for ompl problem generator
  - [ ] Add unit test for trajopt problem generator
  - [ ] Add unit test for descartes problem generator

### CODE REVIEW

**Comment by Levi-Armstrong** [comment]
@marrts and @mpowelson Do either of you have time to take a look at my commits that I added to mpowelson? I think we can get this merged and move the remaining items to an issue.

File: tesseract/tesseract_planning/tesseract_motion_planners/test/CMakeLists.txt [hunk 1]
```diff
@@ -35,19 +35,38 @@ add_dependencies(run_tests ${PROJECT_NAME}_ompl_unit)
...
 add_executable(${PROJECT_NAME}_trajopt_unit trajopt_planner_tests.cpp)
-target_link_libraries(${PROJECT_NAME}_trajopt_unit PRIVATE GTest::GTest GTest::Main ${PROJECT_NAME}_trajopt tesseract::tesseract_support)
+target_link_libraries(${PROJECT_NAME}_trajopt_unit PRIVATE GTest::GTest GTest::Main ${PROJECT_NAME}_trajopt tesseract::tesseract_support ${PROJECT_NAME}_simple)
 tesseract_target_compile_options(${PROJECT_NAME}_trajopt_unit PRIVATE)
 tesseract_clang_tidy(${PROJECT_NAME}_trajopt_unit)
 tesseract_code_coverage(${PROJECT_NAME}_trajopt_unit ALL EXCLUDE ${COVERAGE_EXCLUDE})
 tesseract_gtest_discover_tests(${PROJECT_NAME}_trajopt_unit)
 add_dependencies(${PROJECT_NAME}_trajopt_unit ${PROJECT_NAME}_trajopt)
-add_dependencies(run_tests ${PROJECT_NAME}_trajopt_unit)
+#add_dependencies(run_tests ${PROJECT_NAME}_trajopt_unit)
```
Comment (marrts) [inline tesseract/tesseract_planning/tesseract_motion_planners/test/CMakeLists.txt:65]: Are these tests intentionally omitted for now?

File: tesseract/tesseract_planning/tesseract_command_language/include/tesseract_command_language/command_language_utils.h [hunk 2]
```diff
@@ -130,6 +130,8 @@ long getMoveInstructionsCount(const CompositeInstruction& composite_instruction,
  */
 long getPlanInstructionsCount(const CompositeInstruction& composite_instruction, bool process_child_composites = true);
 
+const Eigen::VectorXd& getJointPosition(const Waypoint& waypoint);
```
[the remark this answers is not in the document]
Comment (Levi-Armstrong) [reply tesseract/tesseract_planning/tesseract_command_language/include/tesseract_command_language/command_language_utils.h:133]: Done

File: tesseract/tesseract_planning/tesseract_command_language/src/command_language_utils.cpp [hunk 3]
```diff
@@ -31,12 +31,19 @@ TESSERACT_COMMON_IGNORE_WARNINGS_POP
...
 
 namespace tesseract_planning
 {
 const PlanInstruction* getFirstPlanInstruction(const CompositeInstruction& composite_instruction,
                                                bool process_child_composites)
 {
+  if (composite_instruction.hasStartInstruction())
+    if (isPlanInstruction(composite_instruction.getStartInstruction()))
+      return composite_instruction.getStartInstruction().cast_const<PlanInstruction>();
+
```
[the remark this answers is not in the document]
Comment (Levi-Armstrong) [reply tesseract/tesseract_planning/tesseract_command_language/src/command_language_utils.cpp:46]: Fixed documentation.

File: tesseract/tesseract_planning/tesseract_motion_planners/test/CMakeLists.txt (hunk 1 above)
Comment (Levi-Armstrong) [reply tesseract/tesseract_planning/tesseract_motion_planners/test/CMakeLists.txt:65]: I missed it. It is fixed in the latest changes.

**Comment by Levi-Armstrong** [comment]
@marrts You good with me merging this?

**Comment by marrts** [comment]
Yes

**Pull request merged**